corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -85,11 +85,30 @@ EntryArrayAnswer.prototype.onPreProcess = function(newValue) {
  */
 EntrySingleAnswer = function(question, options) {
     var self = this;
+
     Entry.call(self, question, options);
     if (question.answer()) {
-        self.rawAnswer = ko.observable(question.answer());
+        if (options.enableRateLimit) {
+            self.rawAnswer = ko.observable(question.answer()).extend({
+                rateLimit: {
+                    timeout: 500,
+                    method: "notifyWhenChangesStop"
+                }
+            });
+        } else {
+            self.rawAnswer = ko.observable(question.answer());
+        }
     } else {
-        self.rawAnswer = ko.observable(Formplayer.Const.NO_ANSWER);
+        if (options.enableRateLimit) {
+            self.rawAnswer = ko.observable(Formplayer.Const.NO_ANSWER).extend({
```

Comment:
i think this whole thing can be condensed if we do:
```
 self.rawAnswer = ko.observable(self.answer() || Formplayer.Const.NO_ANSWER).extend(...
```
i know it wasn't you originally, but now it's a significant amount of code copying